crates/oxide/src/parser.rs

```diff
@@ -303,6 +307,14 @@ impl<'a> Extractor<'a> {
             }
         }
 
+        // CSS variables must be preceded by `var(` to be considered a valid CSS variable candidate
+        if candidate.starts_with(b"--") {
+            match input.get(start_idx - 4..start_idx) {
+                Some(b"var(") => return ValidationResult::Valid,
+                _ => return ValidationResult::Invalid,
+            }
+        }
```

Comment:
I think conceptually it would be better to not mix the two concept of candidates and variables names in the parsing code since it's already quite convoluted here... I wonder if this is the time where we go back to the drawing board and try to make this a bit easier to understand?